Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/FilterAdapter.cpp

```diff
@@ -0,0 +1,501 @@
+                    {
+                        // this value wasn't a row before, so its parent needs to recache its info
+                        auto parentNode = GetMatchNodeAtPath(patchPath)->m_parentNode;
+                        CacheDomInfoForNode(sourceContents[patchPath].ArrayPopBack(), parentNode);
+                        UpdateMatchState(parentNode);
+                    }
+                }
+                else
+                {
+                    // not a row now. Check if it was before
+                    if (existingMatchInfo)
+                    {
+                        // node being replaced was a row. We need to remove it from our graph
+                        auto& parentContainer = existingMatchInfo->m_parentNode->m_childMatchState;
+                        parentContainer.erase(parentContainer.begin() + (patchPath.end() - 1)->GetIndex());
+
+                        // update former parent's match state, as its m_matchingDescendents may have changed
+                        UpdateMatchState(existingMatchInfo->m_parentNode);
+                    }
+                    else
+                    {
+                        // replaced node wasn't a row, so we need to re-cache the owning row's cached info
+                        auto matchingRow = GetMatchNodeAtPath(rowPath);
+                        CacheDomInfoForNode(sourceContents[rowPath], matchingRow);
+                        UpdateMatchState(matchingRow);
+                    }
+                }
+
+            }
+            else if (operationIterator->GetType() == AZ::Dom::PatchOperation::Type::Add)
+            {
+                // <apm> test this! Does an add imply shifting all subsequent nodes back by 1, or are we given those messages?
+                // i.e., should we insert a blank m_childMatchState entry before we populate here?
+                if (rowPath == patchPath)
+                {
+                    // given path is a new row, populate our tree with match information for this node and any row descendents
+                    PopulateNodesAtPath(rowPath);
+                }
+                else
+                {
+                    // added node wasn't a row, so we need to re-cache the owning row's cached info
+                    auto matchingRow = GetMatchNodeAtPath(rowPath);
+                    CacheDomInfoForNode(sourceContents[rowPath], matchingRow);
+                    UpdateMatchState(matchingRow);
+                }
+            }
+            else
+            {
+                AZ_Error("FilterAdapter", false, "patch operation not supported yet");
+            }
+        }
+        NotifyResetDocument();
+    }
+
+    void RowFilterAdapter::HandleDomMessage(const AZ::DocumentPropertyEditor::AdapterMessage& message,[[maybe_unused]] Dom::Value& value)
+    {
+        // forward all messages unaltered
+        #undef SendMessage // for now, work around windows #defines. We should probably just rename our SendMessage
```

Comment:
+1 to renaming the method to not cause pain - this also might need an `#ifdef` for non-Windows platforms but I'm sure AR will be happy to weigh in

Reply:
renamed to SendAdapterMessage and undef removed